### PR TITLE
Add: "Integrating Lambda Chat into VS Code"

### PR DESCRIPTION
**Pull request by cbrownstein-lambda**
## Description

This PR adds a programming guide, _Integrating Lambda Chat into VS Code_.

## Related issues

- [[CEX-16](https://lambdalabs.atlassian.net/browse/CEX-16)] (JIRA)

## Checklist

- [x] Documentation follows the style guidelines of the project.
- [x] Files used in this PR (e.g., images and PDFs) are committed.
- [x] Link to existing documentation and the reverse.
- [x] Changes have been tested locally.
- [x] Reviewers have been assigned (at least 1 tech writer and 1 eng).

## Additional notes

None

### CODE REVIEW

**Review by smithbrady — COMMENTED** [review]
It all looks great! Didn't see any issues.

**Comment by cbrownstein-lambda** [comment]
I'm thinking about replacing the example with a "find problems" example.

Screenshot of CodeGPT: Find Problems example

File: docs/education/programming/vs-code-lambda-chat.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  - llm
+  - programming
+---
+
+# Integrating Lambda Chat into VS Code
+
+## Introduction
+
+Using the
+[CodeGPT VS Code (Visual Studio Code) extension :octicons-link-external-16:](https://marketplace.visualstudio.com/items?itemName=DanielSanMedium.dscodegpt){target="_blank"},
```
Comment (pete-lambda) [inline docs/education/programming/vs-code-lambda-chat.md:16]: Global: Do we have to manually enter the octicon here?

File: docs/education/programming/vs-code-lambda-chat.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+```
+ext install DanielSanMedium.dscodegpt
+```
+
+Then, press ++enter++ to install the CodeGPT extension.
+
+## Configure the Lambda Chat integration
+
+In the Primary Side Bar, click the **CHATGPT** button.
```
Comment (pete-lambda) [inline docs/education/programming/vs-code-lambda-chat.md]: I see one instance of "primary side bar" in their help and one instance of it capitalized. I think lowercase reads easier, plus it doesn't have a proper name in the UI from what I can tell. 

Also, MS MoS uses it as one word: https://learn.microsoft.com/en-us/style-guide/scannable-content/sidebars

File: docs/education/programming/vs-code-lambda-chat.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+in code, and much more.
+[See CodeGPT's documentation to learn more :octicons-link-external-16:](https://docs.codegpt.co/docs/intro){target="_blank"}.
+
+For this tutorial, you need to
+[generate a Cloud API key :octicons-link-external-16:](https://cloud.lambdalabs.com/api-keys){target="_blank"}.
+You can also use an existing Cloud API key.
+
+## Install the CodeGPT extension
+
+First, open VS Code.
```
Comment (pete-lambda) [inline docs/education/programming/vs-code-lambda-chat.md:29]: Number these steps?

File: docs/education/programming/vs-code-lambda-chat.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+Press ++ctrl++ + ++p++ to open **Quick Open**. Enter the following:
+
+```
+ext install DanielSanMedium.dscodegpt
+```
+
+Then, press ++enter++ to install the CodeGPT extension.
+
+## Configure the Lambda Chat integration
+
```
Comment (pete-lambda) [inline docs/education/programming/vs-code-lambda-chat.md:40]: Number these steps?

File: docs/education/programming/vs-code-lambda-chat.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+Then, click **Connect**.
+
+You should see:
+
+![CODEGPT connected](../../assets/images/codegpt-connected.png)
+
+## Test the Lambda Chat integration
+
+In the menu bar, select **File** > **New Text File**.
```
Comment (pete-lambda) [inline docs/education/programming/vs-code-lambda-chat.md:78]: Number these steps?

File: docs/education/programming/vs-code-lambda-chat.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+Enter a snippet of code in the editor, for example:
+
+```python
+for _ in range(3):
+    print("Hello, World!")
+```
+
+Select the code, then right-click the code. Choose **CodeGPT: Explain**.
+
+Near the bottom of the Primary Side Bar, click **Send**.
```
Comment (pete-lambda) [inline docs/education/programming/vs-code-lambda-chat.md:89]: Same as above wrt "primary side bar"

**Review by pete-lambda — CHANGES_REQUESTED** [review]
Looks good. Left some feedback in there, mostly to number steps and whether we should use Primary Side Bar or primary side bar.

File: docs/education/programming/vs-code-lambda-chat.md (hunk 1 above)
Comment (cbrownstein-lambda) [reply docs/education/programming/vs-code-lambda-chat.md:16]: In Vim I have that and `{target="_blank"}` in registers so I can add them with 3 key presses.

I think [snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets) in VS Code are similar.

File: docs/education/programming/vs-code-lambda-chat.md (hunk 2 above)
Comment (cbrownstein-lambda) [reply docs/education/programming/vs-code-lambda-chat.md]: In **View** > **Appearance** it's **Primary Side Bar** (capitalized, two words). In the [Extension API docs](https://code.visualstudio.com/api/ux-guidelines/sidebars) it's **Primary Sidebar** (capitalized, one word).

I suggest we go with **Primary Sidebar** (capitalized, one word).

File: docs/education/programming/vs-code-lambda-chat.md (hunk 6 above)
Comment (cbrownstein-lambda) [reply docs/education/programming/vs-code-lambda-chat.md:89]:
```suggestion
Near the bottom of the Primary Sidebar, click **Send**.
```

File: docs/education/programming/vs-code-lambda-chat.md (hunk 3 above)
Comment (cbrownstein-lambda) [reply docs/education/programming/vs-code-lambda-chat.md]: Steps numbered in bed47fbae338cdd82f749449866ff701b84b6bdb.

File: docs/education/programming/vs-code-lambda-chat.md (hunk 1 above)
Comment (pete-lambda) [reply docs/education/programming/vs-code-lambda-chat.md:16]: I see. Would be cool if we could just program that into MkDocs so external links would get the icon and always go into a new window.

**Review by pete-lambda — APPROVED** [review]
Looks great now. And that's fine about Primary Sidebar.

**Pull request merged**